documentation-site/components/survey.js

```diff
@@ -0,0 +1,148 @@
+    }, delay);
+  }, []);
+
+  return isOpen ? (
+    <div
+      className={useCss({
+        position: 'fixed',
+        bottom: 0,
+        right: 0,
+        left: 0,
+        width: '100%',
+        boxShadow: theme.lighting.shadow400,
+        textAlign: 'center',
+        paddingTop: theme.sizing.scale800,
+        paddingBottom: theme.sizing.scale800,
+        backgroundColor: theme.colors.background,
+        [`@media screen and (max-width: ${theme.breakpoints.medium}px`]: {
+          display: 'none',
+        },
+      })}
+    >
+      <div
+        className={useCss({
+          marginBottom: theme.sizing.scale400,
+        })}
+      >
+        <Label2>Overall, how satisfied are you with Base Web?</Label2>
+      </div>
+      <StarRating
+        numItems={5}
+        onChange={data => {
+          setRating(data.value);
+          trackEvent('survey', 'csat', data.value);
+          window.open(`
+            https://docs.google.com/forms/d/e/1FAIpQLSfQ6uxhW96LX31x5hbp2xIb-WPI0eIavfAb_s7lkhKnz-LIJQ/viewform?usp=pp_url&entry.1493106267=${data.value}
+          `);
+          setIsOpen(false);
+        }}
+        value={rating}
+      />
+      <div
+        role="button"
+        tabIndex="0"
+        onKeyPress={event => {
+          if (event.key === 'Enter') {
+            setIsOpen(false);
+          }
+        }}
+        className={useCss({
+          cursor: 'pointer',
+          position: 'absolute',
+          top: theme.sizing.scale400,
+          right: theme.sizing.scale400,
+        })}
+        onClick={() => {
+          setIsOpen(false);
+        }}
+      >
+        <Delete size="24px" />
+      </div>
+    </div>
+  ) : null;
+};
+
+// we want to show the survey once every two months
+function shouldShowSurvey() {
+  try {
+    // forcing the display of the survey for testing purposes
+    const {search} = window.location;
+    if (search && search.includes('survey')) {
+      return {showSurvey: true, delay: 0};
+    }
+  } catch (err) {
+    // do nothing
+  }
+
+  const Cookies = new CookiesConstructor();
+
+  const cookies = {
+    FIRST_SEEN: 'survey-user-first-seen',
+    LAST_SURVEYED: 'survey-last-surveyed',
+  };
+  const TWO_MONTHS_AGO = 60 * 24 * 60 * 60 * 1000;
+  const ONE_WEEK_AGO = 7 * 24 * 60 * 60 * 1000;
+  const SURVEY_DELAY = 2 * 60 * 1000; // 2 minutes
+
+  const firstSeen = Cookies.get(cookies.FIRST_SEEN);
+  const lastSurveyed = Cookies.get(cookies.LAST_SURVEYED);
+  const delay = 0;
+  const now = new Date();
+
+  // the person is a new visitor, bailing out
+  if (!firstSeen) {
+    Cookies.set(cookies.FIRST_SEEN, now.getTime(), {
+      // by default, cookies are session cookies, so without an expiration they are removed
+      // when the browser window is closed.
+      expires: new Date(now.getFullYear() + 10, now.getMonth()),
+    });
+    return {showSurvey: false, delay};
+  }
+
+  // the person only knows base web for less than a week
+  if (firstSeen > ONE_WEEK_AGO) {
```

Comment:
should this be `<`?

Reply:
(hypothetical numbers)
if now is `200ms` since epoch, one week ago was `100ms`, and first seen is `120ms`: that means that first seen is within the one week range

this conditional could be modeled closer to the comment description as a delta between now and the measured times

Reply:
oh gotcha - at a glance it seems backwards. "first seen over a week ago"

Reply:
hm, let me change things around so it is less confusing

Reply:
@chasestarr @sandgraham just updated the conditional - does it make more sense now?

Reply:
don't see an update, push the change?

Reply:
oh lol, just left my terminal:

```
giut: command not found
```

should be updated now